exchange/events.go

```diff
@@ -53,7 +54,7 @@ func (ev *eventTracking) modifyBidsForEvents(seatBids map[openrtb_ext.BidderName
 
 // isModifyingVASTXMLAllowed returns true if this bidder config allows modifying VAST XML for event tracking
 func (ev *eventTracking) isModifyingVASTXMLAllowed(bidderName string) bool {
-	return ev.bidderInfos[bidderName].ModifyingVastXmlAllowed && (ev.enabledForAccount || ev.enabledForRequest)
+	return ev.bidderInfos[bidderName].ModifyingVastXmlAllowed && ((ev.enabledForAccount != nil && *ev.enabledForAccount) || ev.enabledForRequest || ev.enabledForInstance)
```

Comment:
Under the current logic, if `ev.enabledForAccount` is `false` and `ev.enabledForInstance` is `true`, the `ev.enabledForInstance` `true` value prevails despite being lower in the order of precedence @bretg mentioned in issue #2479:
1. request-level params
2. account-level config
3. host-level config
4. hardcoded defaults

I believe we should swap `ev.enabledForInstance` to the forefront of the second part of the `&&` operand and and make sure `ev.enabledForAccount` equals `nil` so the short-circuit OR doesn't omit an account override. In other words:
```
55   // isModifyingVASTXMLAllowed returns true if this bidder config allows modifying VAST XML for event tracking
56   func (ev *eventTracking) isModifyingVASTXMLAllowed(bidderName string) bool {
57 -     return ev.bidderInfos[bidderName].ModifyingVastXmlAllowed && ((ev.enabledForAccount != nil && *ev.enabledForAccount) || ev.enabledForRequest || ev.enabledForInstance)
   +     return ev.bidderInfos[bidderName].ModifyingVastXmlAllowed && ((ev.enabledForAccount == nil && ev.enabledForInstance) || (ev.enabledForAccount != nil && *ev.enabledForAccount) || ev.enabledForRequest)
58   }
exchange/events.go
```
Also, given that this boolean expression is used in three different places, can we put it in it's own function and write a table-driven test case for it?
```
55   // isModifyingVASTXMLAllowed returns true if this bidder config allows modifying VAST XML for event tracking
56   func (ev *eventTracking) isModifyingVASTXMLAllowed(bidderName string) bool {
57 -     return ev.bidderInfos[bidderName].ModifyingVastXmlAllowed && ((ev.enabledForAccount != nil && *ev.enabledForAccount) || ev.enabledForRequest || ev.enabledForInstance)
   +     return ev.bidderInfos[bidderName].ModifyingVastXmlAllowed && isEventAllowed(ev.enabledForRequest, ev.enabledForAccount, ev.enabledForInstance)
58   }
   + 
   + func isEventAllowed(requestEnabled bool, accountEnabled *bool, hostEnabled bool) bool {
   +     return (accountEnabled == nil && hostEnabled) ||  (accountEnabled != nil && *accountEnabled) || requestEnabled
   + }
exchange/events.go
```